### PR TITLE
VB-717 Conflicts added to available sessions response

### PR DESCRIPTION
**Pull request by philip-milne**
## What does this pull request do?
Adds additional information to the available sessions response.
Adds conflict filtering configuration to the application config.

## What is the intent behind these changes?
Provide the client with additional information about the available sessions.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt [hunk 1]
```diff
@@ -262,31 +271,43 @@ class SessionServiceTest {
...
+        clock,
+        1,
+        100,
+        policyFilterDoubleBooking = false,
+        policyFilterNonAssociation = false,
+        policyNonAssociationWholeDay = true,
+      )
+    }
 
     private fun mockRepositoryResponse(response: List<SessionTemplate>) {
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:292]: better naming I think I don't know what mockRepositoryResponse

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt [hunk 2]
```diff
@@ -391,37 +406,84 @@ class SessionServiceTest {
...
       // When
       val sessions = sessionService.getVisitSessions(prisonId, prisonerId)
 
       // Then
-      assertThat(sessions).size().isEqualTo(0)
+      assertThat(sessions).size().isEqualTo(1)
+      assertDate(sessions[0].startTimestamp, "2021-01-02T11:30:00", SATURDAY)
+      assertThat(sessions[0].sessionConflicts).size().isEqualTo(1)
+      assertThat(sessions[0].sessionConflicts!!.first()).isEqualTo(SessionConflict.NON_ASSOCIATION)
       Mockito.verify(prisonApiClient, times(1)).getOffenderNonAssociation(prisonerId)
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:429]: nice

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt [hunk 3]
```diff
@@ -391,37 +406,84 @@ class SessionServiceTest {
...
+        )
+
+      // When
+      val sessions = sessionService.getVisitSessions(prisonId, prisonerId)
+
+      // Then
+      assertThat(sessions).size().isEqualTo(1)
+      assertDate(sessions[0].startTimestamp, "2021-01-02T11:30:00", SATURDAY)
+      assertThat(sessions[0].sessionConflicts).size().isEqualTo(1)
+      assertThat(sessions[0].sessionConflicts!!.first()).isEqualTo(SessionConflict.DOUBLE_BOOKED)
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:474]: Nice

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt [hunk 4]
```diff
@@ -391,37 +406,84 @@ class SessionServiceTest {
...
+          emptyList()
         )
 
       // When
       val sessions = sessionService.getVisitSessions(prisonId, prisonerId)
 
       // Then
-      assertThat(sessions).size().isEqualTo(0)
+      assertThat(sessions).size().isEqualTo(1)
+      assertDate(sessions[0].startTimestamp, "2021-01-02T11:30:00", SATURDAY)
```
Comment (AledWEvans) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:426]: I do hate this startTimestamp, but not for this change/review

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -95,7 +95,10 @@ prison:
 policy:
   session:
     non-association:
+      filter: true
       whole-day: true
+    double-booking:
+      filter: false
```
Comment (AledWEvans) [inline src/main/resources/application.yml:101]: is it false? I suppose we are turning it off until a later date! Do we have to put something in for helm?

Comment (philip-milne) [reply src/main/resources/application.yml:101]: The requirements might change post mvp, config added to allow the UI to change without re-coding the api.  This config can be overridden, defaults for now to the current requirements.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt (hunk 1 above)
Comment (philip-milne) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionServiceTest.kt:292]: same

**Review by AledWEvans — APPROVED** [review]

**Pull request merged**